src/unicode.cpp

```diff
@@ -108,9 +99,9 @@ namespace ada::unicode {
 
     for (uint8_t iterator: input) {
       if (character_sets::bit_at(character_set, iterator)) {
-        result += character_sets::hex + iterator * 4;
+        result.append(character_sets::hex + iterator * 4);
```

Comment:
The same is true here, you are using standard C++ call that *may* throw, inside a noexcept function. That is a bit controversial.

Reply:
Is there a performance benefit for using `append` instead of `+=`?

Reply:
You'd want to use neither, honestly.